linux/gnome/src/main/cpp/GnomeThemeUtils.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Jannis Weis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+#include "com_github_weisj_darkmode_platform_linux_gnome_GnomeThemeUtils.h"
+
+#include <glibmm-2.4/glibmm.h>
+#include <giomm-2.4/giomm.h>
+#include <gtk/gtk.h>
+#include <utility>
+
+bool gioAlreadyInit = false;
+
+void ensure_gio_init() {
```

Comment:
This should probably be in sync with `DarkModeGnome.cpp`. May be move the initialisation of gio into an own file, so both other files can use it?

Reply:
Fixed with 69675c4051f185b4c114b23f451a6e3e987d3e98.